lib/Test/MockFile.pm

```diff
@@ -1712,6 +1714,120 @@ BEGIN {
         $mock->{'has_content'} = undef;
         return 1;
     };
+
+    *CORE::GLOBAL::chown = sub(@) {
+        my ( $uid, $gid, @files ) = @_;
+
+        $^O eq 'MSWin32'
+            and return 0; # does nothing on Windows
+
+        # Not an error, report we changed zero files
+        @files
+            or return 0;
+
+        my %mocked_files = map +( $_ => _get_file_object($_) ), @files;
+
+        # The idea is that if some are mocked and some are not,
+        # it's probably a mistake
+        my $mocked_count = grep ref, values %mocked_files;
+        if ( $mocked_count && $mocked_count != @files ) {
+            die "You called chown() on a mix of mocked and unmocked files - this is very likely a bug on your side";
+        }
+
+        my $num_changed = 0;
+        foreach my $file (@files){
+            my $mock = $mocked_files{$file};
+
+            if ( !$mock ) {
+                _real_file_access_hook( 'chown', \@_ );
+                goto \&CORE::chown if _goto_is_available();
+                return CORE::chown(@files);
+            }
+
+            if ( !$mock->exists() ) {
+                if ( $mock->is_file() ) {
+                    $! = ENOENT;
```

Comment:
does $! really clobber like this for a list of files? or does it fail on the first one?

Reply:
I imagine the clobbering is certainly wrong. Either it should only clobber with errors (so a non-error would not obscure an error) or it shouldn't clobber at all. I'll check and revise. Thanks!